Xwt.XamMac/Xwt.Mac/MacEngine.cs

```diff
@@ -155,6 +155,14 @@ public override void ExitApplication ()
 			NSApplication.SharedApplication.Terminate(appDelegate);
 		}
 
+		public override void ExitApplication(int exitCode)
+		{
+			if (exitCode == 0)
+				NSApplication.SharedApplication.Terminate(appDelegate);
+			else
+				Environment.Exit(exitCode);
```

Comment:
This is not 100% correct. Using Environment.Exit will almost always be a slow quit.

Reply:
Just wanted to mention this, since last time I tried it, it took 10 seconds to quit

Reply:
Is there any other way of reporting the exit code?

Reply:
https://developer.apple.com/documentation/appkit/nsapplication/1428417-terminate?language=objc

Not that I can find.

Also, calling Environment.Exit will prevent normal ShouldTerminate/WillTerminate handlers from running, afaik.

Given app bundles are supposed to be run by launch services, not directly launched, an exit code does not make sense, as it's not observable, I guess?

Reply:
We need it since we need a result when running unit tests through vstool